dish_grpc_influx2.py

```diff
@@ -0,0 +1,330 @@
+BULK_MEASUREMENT = "spacex.starlink.user_terminal.history"
+FLUSH_LIMIT = 6
+MAX_BATCH = 5000
+MAX_QUEUE_LENGTH = 864000
+
+
+class Terminated(Exception):
+    pass
+
+
+def handle_sigterm(signum, frame):
+    # Turn SIGTERM into an exception so main loop can clean up
+    raise Terminated
+
+
+def parse_args():
+    parser = dish_common.create_arg_parser(output_description="write it to an InfluxDB 2.0 database")
+
+    group = parser.add_argument_group(title="InfluxDB 2.0 database options")
+    group.add_argument("-u",
+                       "--url",
+                       default=HOST_URL,
+                       dest="url",
+                       help="URL or IP AND Port of the InfluxDB 2.0 Host, default: " + HOST_URL)
+    group.add_argument("-T", "--token", help="Token to access the bucket")
+    group.add_argument("-B",
+                       "--bucket",
+                       default=BUCKET_DEFAULT,
+                       help="Bucket name to use, default: " + BUCKET_DEFAULT)
+    group.add_argument("-O", "--org", help="Organisation name")
+    group.add_argument("-k",
+                       "--skip-query",
+                       action="store_true",
+                       help="Skip querying for prior sample write point in bulk mode")
+    group.add_argument("-C",
+                       "--ca-cert",
+                       dest="verify_ssl",
+                       help="Enable SSL/TLS using specified CA cert to verify broker",
+                       metavar="FILENAME")
+    group.add_argument("-I",
+                       "--insecure",
+                       action="store_false",
+                       dest="verify_ssl",
+                       help="Enable SSL/TLS but disable certificate verification (INSECURE!)")
+    group.add_argument("-S",
+                       "--secure",
+                       action="store_true",
+                       dest="verify_ssl",
+                       help="Enable SSL/TLS using default CA cert")
+
+    env_map = (
+        ("INFLUXDB_URL", "url"),
+        ("INFLUXDB_TOKEN", "token"),
+        ("INFLUXDB_Bucket", "bucket"),
+        ("INFLUXDB_ORG", "org"),
+        ("INFLUXDB_SSL", "verify_ssl"),
+    )
+    env_defaults = {}
+    for var, opt in env_map:
+        # check both set and not empty string
+        val = os.environ.get(var)
+        if val:
+            if var == "INFLUXDB_SSL" and val == "secure":
+                env_defaults[opt] = True
+            elif var == "INFLUXDB_SSL" and val == "insecure":
+                env_defaults[opt] = False
+            else:
+                env_defaults[opt] = val
+    parser.set_defaults(**env_defaults)
+
+    opts = dish_common.run_arg_parser(parser, need_id=True)
+
+    opts.icargs = {}
+    for key in ["url", "token", "bucket", "org", "verify_ssl"]:
+        val = getattr(opts, key)
+        if val is not None:
+            opts.icargs[key] = val
+
+    if opts.verify_ssl is not None:
+        opts.icargs["ssl"] = True
```

Comment:
Another API change here, I think. kwarg "ssl" appears to be ignored. Probably needs to be encoded into URL instead.